src/ng/browser.js

```diff
@@ -131,6 +131,15 @@ function Browser(window, document, $log, $sniffer) {
     // setter
     if (url) {
       var sameState = lastHistoryState === state;
+      var hasTrailingSlash = url[url.length - 1] === '/';
+
+      // Normalize the inputted URL
+      url = urlResolve(url).href;
```

Comment:
I think this is actually wrong, and I'm surprised it hasn't caused issues. This `url` should be the full url, that is protocol+domain+path+params+hash, basically equivalent to `window.location.href`...